server/odc-service/src/main/java/com/oceanbase/odc/service/schedule/flowtask/ResultSetExportPreprocessor.java

```diff
@@ -53,15 +53,15 @@ public void process(CreateFlowInstanceReq req) {
         ConnectionConfig connectionConfig =
                 connectionService.getForConnectionSkipPermissionCheck(req.getConnectionId());
 
-        String sql = parameter.getSql();
-        preCheckSql(sql, connectionConfig.getDialectType());
+        preCheckSql(parameter, connectionConfig.getDialectType());
 
         PreConditions.validArgumentState(FILE_NAME_PATTERN.matcher(parameter.getFileName()).matches(),
                 ErrorCodes.BadArgument, null,
                 "File name must contain only letters, numbers, Chinese characters and \"._-\" ");
     }
 
-    private void preCheckSql(String sql, DialectType dialectType) {
+    private void preCheckSql(ResultSetExportTaskParameter parameter, DialectType dialectType) {
+        String sql = parameter.getSql();
```

Comment:
is the new code different from old version?

Reply:
I added `parameter.setSql(sqls.get(0));` at line 80.